File: diff_drive_control/src/circle.cpp
#include<ros/ros.h>
#include<geometry_msgs/Twist.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "circle_node");
    ros::NodeHandle n;
    ros::Publisher pub = n.advertise<geometry_msgs::Twist>("/cmd_vel",1000);
    ros::Rate rate(10);
    ROS_INFO("Moving the bot.");
    int radius = 1;
    while(ros::ok())
    {
        geometry_msgs::Twist vel;
        vel.linear.x = 0.4;
        vel.angular.z = 0.4/radius;
        pub.publish(vel);
        ros::spinOnce();
        rate.sleep();
    }
    ROS_INFO("Stopping the bot."); 
    return 0;
}

File: diff_drive_control/scripts/circle.py
#!/usr/bin/env python

import rospy
from geometry_msgs.msg import Twist

def circle():
    rospy.init_node('circle_node', anonymous=True)
    pub = rospy.Publisher('/cmd_vel', Twist, queue_size=10)
    velocity = Twist()
    rate = rospy.Rate(10)
    rospy.loginfo("Moving the bot")
    radius = 1

    while not rospy.is_shutdown():
        velocity.linear.x = 0.4
        velocity.angular.z = 0.4/radius
        pub.publish(velocity)
        rate.sleep()
    
    velocity.linear.x = 0
    velocity.angular.z = 0
    pub.publish(velocity)

if __name__=='__main__':
    try:
        circle()
    except rospy.ROSInterruptException:
        pass